dvc/repo/__init__.py

```diff
@@ -157,7 +157,7 @@ def collect(self, target, with_deps=False, recursive=False):
         import networkx as nx
         from dvc.stage import Stage
 
-        if not target or recursive:
+        if not target or (recursive and os.path.isdir(target)):
```

Comment:
just curious what would happen it target is a symlink to a dir?

Reply:
os.path.isdir will return True for symlink to directory, so I think it should behave as intended... If symlink is under project root

Reply:
isdir will return True